mondrian/src/main/java/mondrian/olap/QueryTiming.java

```diff
@@ -210,17 +174,7 @@ private void markEnd(long tstamp) {
             this.endTime = tstamp;
         }
     }
-
-    private static class StartEnd {
-        final long startTime;
-        final long endTime;
-
-        public StartEnd(long startTime, long endTime) {
-            this.startTime = startTime;
-            this.endTime = endTime;
-        }
-    }
-
+    
     private static class DurationCount {
```

Comment:
I'm not sure this object can be used in a HashMap as you did. It does not implement hashCode, nor equals, so the map above will be based on identity, thus removing all caching advantages might have been. I think it needs equals and hashCode implemented correctly.

Reply:
The DurationCount object is used as the HashMap value and not the key, so it doesn't need to implement hashCode or Equals.  BTW, I removed StartEnd and re-used DurationCount because StartEnd was creating way too many objects if a function was repeatedly called.  (Ex. a 1,000 tuple row axis with a calculated measure that aggregated on 1,000 members that also involved the aggregate function could result in 1M AggregateFunDef calls and 1M of these StartEnd objects).

Reply:
My bad. I read this too fast.